src/scrapers/visa-cal.ts

```diff
@@ -326,12 +331,11 @@ class VisaCalScraper extends BaseScraperWithBrowser<ScraperSpecificCredentials>
     return initData?.result.cards.map(({ cardUniqueId, last4Digits }) => ({ cardUniqueId, last4Digits }));
   }
 
-  async getAuthorizationHeader() {
-    const authModule = await getFromSessionStorage<{ auth: { calConnectToken: string } }>(this.page, 'auth-module');
-    if (!authModule) {
-      throw new Error('could not find \'auth-module\' in session storage');
+  getAuthorizationHeader() {
+    if (!this.authorization) {
+      throw new Error('could not retrieve authorization header');
```

Comment:
Considre falling back to the existing method

```suggestion
  async getAuthorizationHeader() {
    if (!this.authorization) {
      const authModule = await getFromSessionStorage<{ auth: { calConnectToken: string | null } }>(this.page, 'auth-module');
      if (authModule.auth.calConnectToken) {
          `CALAuthScheme ${authModule.auth.calConnectToken}`;
      }
      throw new Error('could not retrieve authorization header');
```